opba-protocols/hbci-protocol/src/main/java/de/adorsys/opba/protocol/hbci/service/consent/authentication/HbciSendTanChallenge.java

```diff
@@ -20,21 +21,43 @@
  */
 @Service("hbciSendTanChallenge")
 @RequiredArgsConstructor
+@Slf4j
 public class HbciSendTanChallenge extends ValidatedExecution<HbciContext> {
 
     private final OnlineBankingService onlineBankingService;
+    private final HbciAuthorizationPossibleErrorHandler errorSink;
 
     @Override
     protected void doRealExecution(DelegateExecution execution, HbciContext context) {
-        HbciConsent consent = context.getHbciDialogConsent();
 
-        TransactionAuthorisationRequest request = create(new BankApiUser(), new BankAccess(), context.getBank(), consent);
-        request.setScaAuthenticationData(context.getPsuTan());
+        errorSink.handlePossibleAuthorizationError(
+                () -> {
+                    context.setWrongAuthCredentials(false);
```

Comment:
Suggest to extract (Runnable) lambda function body to the private method as it is quite long
It is a smell as well as causes problems handling non-runtime exceptions:
https://developer.ibm.com/articles/j-java8idioms6/

Reply:
done. Thanks for link!